internal/venice-client-common/src/main/java/com/linkedin/venice/stats/routing/HelixGroupStats.java

```diff
@@ -67,4 +151,87 @@ public double getGroupResponseWaitingTimeAvg(int groupId) {
     }
     return avgLatency;
   }
+
+  /**
+   * Helper class to hold OpenTelemetry setup data for a specific helix group.
+   */
+  private static class OtelSetupData {
+    final VeniceOpenTelemetryMetricsRepository otelRepository;
+    final Map<VeniceMetricsDimensions, String> baseDimensionsMap;
+    final Attributes baseAttributes;
+
+    OtelSetupData(
+        VeniceOpenTelemetryMetricsRepository otelRepository,
+        Map<VeniceMetricsDimensions, String> baseDimensionsMap,
+        Attributes baseAttributes) {
+      this.otelRepository = otelRepository;
+      this.baseDimensionsMap = baseDimensionsMap;
+      this.baseAttributes = baseAttributes;
+    }
+  }
+
+  /**
+   * Builds OpenTelemetry setup data for a specific helix group.
+   * Returns null values if storeName is empty (Router mode).
+   */
+  private OtelSetupData buildOtelSetupForGroup(int groupId) {
```

Comment:
what would the groupid values be?
I am not really liking this, but better than having group id as a dynamic variable and create a MES for that.

Reply:
Yeah, I wanted to avoid using `MetricEntityStateGeneric` if possible and in this way, we could achieve it with some complexity. What do you think?